src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -634,7 +664,67 @@ - (XCTestExpectation *)announceProviderToDevice:(MTRDevice *)device
     return responseExpectation;
 }
 
-- (void)test001_ReceiveOTAQuery
+/**
+ * Helper that checks if the OTA provider locations match.
+ */
+- (bool)AreEqual:(MTROtaSoftwareUpdateRequestorClusterProviderLocation *)pl1
+    providerLocation:(MTROtaSoftwareUpdateRequestorClusterProviderLocation *)pl2
+{
+    return [pl1.providerNodeID isEqualToNumber:pl2.providerNodeID] && [pl1.endpoint isEqualToNumber:pl2.endpoint] &&
+        [pl1.fabricIndex isEqualToNumber:pl2.fabricIndex];
+}
+
+- (void)test001_UpdateDefaultOTAProviderAndVerify
+{
+    // Test to update the default OTA provider for a device and verify that it's updated correctly.
+
+    MTROtaSoftwareUpdateRequestorClusterProviderLocation * providerLocation =
+        [[MTROtaSoftwareUpdateRequestorClusterProviderLocation alloc] init];
+    providerLocation.providerNodeID = @(kOTAProviderNodeId);
+    providerLocation.endpoint = @(kOTAProviderEndpointId);
+    providerLocation.fabricIndex = @(kOTAProviderFabricIndex);
+
+    dispatch_queue_t queue = dispatch_get_main_queue();
+    XCTestExpectation * updateDefaultOTAProviderExpectation =
+        [self expectationWithDescription:@"UpdateDefaultOTAProvider verified"];
+
+    [sController
+            getBaseDevice:kDeviceId1
```

Comment:
Please do not use getBaseDevice.  It's deprecated and should never be used.

sConnectedDevice1 exists already, no?

Reply:
yup. i missed that we do have sConnectedDevice1.

Reply:
N/A as I removed this for now